__mongo__/environment.js

```diff
@@ -9,9 +9,11 @@ class MongoEnvironment extends NodeEnvironment {
         await setup();
         await super.setup();
 
-        this.global.testWithNCollections = (n, name, fn) => {
-            this.global.test(name, async () => {
-                const collections = Array.from({length: n}, () => global.__MONGO__.db.collection(uuid()));
+        this.global.testWithNCollections = (newCollections, testName, fn) => {
+            this.global.test(testName, async () => {
+                const collections = Array.from({length: newCollections.length || newCollections}, (_, index) =>
+                    global.__MONGO__.db.collection(newCollections[index] || uuid())
+                );
```

Comment:
Using `Array.isArray(newCollections) ? ... : ...` will make it much easier while adding TypeScript.

Reply:
```ts
                const isArray = Array.isArray(newCollections);
                const collections = Array.from({length: isArray ? newCollections.length : newCollections}, (_, index) =>
                    global.__MONGO__.db.collection(isArray ? newCollections[index] : uuid())
                );
```

Does this look ok or do you have a better idea?

Reply:
This is still not OK for TypeScript, as `isArray: boolean`, not `isArray: assert Array.isArray(newCollections)`. I suggest something like this:

```ts
const db = global.__MONGO__.db;
const collections =
  Array.isArray(newCollections)
    ? newCollections.map(name => db.collection(name))
    ? Array.from({ length: newCollections }, () => db.collection(uuid()));
```

Reply:
Yeah that's what I was also thinking but wasn't sure whether or not `Array.isArray` was a typeguard by itself. Changed.